### PR TITLE
Update examples to use Ember.Helper.helper

### PR DESCRIPTION
**Pull request by tuttinator**
Thanks for all the work done on ember-cli! I really appreciate all the hard work put in.

As per http://guides.emberjs.com/v1.13.0/templates/writing-helpers/#toc_helper-arguments
this commit updates http://www.ember-cli.com/user-guide/#resolving-handlebars-helpers
to use examples for the `Ember.Helper.helper` over the deprecated
`Ember.Handlebars.registerBoundHelper`

The command `ember generate helper foo-bar` also produces this output.

Stylistically this uses destructuring assignment as in the Ember docs example.

I hope this is helpful.

### CODE REVIEW

File: _posts/2014-04-02-using-modules-and-the-resolver.md [hunk 1]
```diff
@@ -129,7 +131,7 @@ explicitly:
 
 {% highlight javascript linenos %}
 // app/helpers/trim.js
-export default function(value, options) {
+export default function([value], options) {
```
Comment (stefanpenner) [inline _posts/2014-04-02-using-modules-and-the-resolver.md]: we don't make babel mandatory yet, `[value]` should be made its ES5 equiv until babel becomes mandatory for JS users

**Comment by bambery** [comment]
Is this the right place to request also updating the docs to remove references to requiring dashes in the helper name, or should I open a new issue?

File: _posts/2014-04-02-using-modules-and-the-resolver.md (hunk 1 above)
Comment (rwjblue) [inline _posts/2014-04-02-using-modules-and-the-resolver.md]: @stefanpenner - This is for the website examples, do we want to continue to use ES5 even in gh-pages?

**Comment by rwjblue** [comment]
@bambery - Please open a new PR for that (definitely welcome though).

**Comment by bambery** [comment]
@tuttinator - you missed one place under "using pods" - "Remember that you must register your helpers by exporting makeBoundHelper or calling registerBoundHelper explicitly."

I'll open a new PR to fix dash references.

**Comment by tuttinator** [comment]
@bambery thanks - just fixed that now

**Comment by szines** [comment]
Thanks @tuttinator, this is what I was looking for. I'm just wondering, will this update be part of the live documentation of ember-cli? What do you think @rwjblue @bambery?

**Comment by stefanpenner** [comment]
cc @leo

**Comment by leo** [comment]
@tuttinator Thank you for your help on improving the docs! But could you please re-target this PR to the [new repo][1] of our site? The old branch has been deleted. Thank you very much! :innocent:

[1]: https://github.com/ember-cli/ember-cli.github.io

**Pull request closed without merging**